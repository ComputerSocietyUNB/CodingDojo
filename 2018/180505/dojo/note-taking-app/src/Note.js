import React, { Component } from 'react'

export default class Note extends Component {
  render() {  return (
    <div>
      <p>
        <input type="text"
          alt="note"
          value={this.props.value}
          onChange={this.props.handleChange}
        />
      </p>
      <button className="button-add-note" onClick={this.props.addNote}>Add Note</button>
    </div>
  );
  }
}
