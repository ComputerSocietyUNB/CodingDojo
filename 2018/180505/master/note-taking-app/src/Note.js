import React, { Component } from 'react'

export default class Note extends Component {
  render() {
    return (
      <div className="Note">
        <textarea
          type="text"
          value={this.props.note}
          onChange={this.props.onNoteEdit}
          placeholder="Your note goes here"
        />
      </div>
    )
  }
}
