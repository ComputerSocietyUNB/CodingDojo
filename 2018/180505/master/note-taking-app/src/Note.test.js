import React from 'react'
import {
  render,
  Simulate,
} from 'react-testing-library'

import Note from './Note'

describe('Note', () => {
  let input

  beforeEach(() => {
    const { container, getByPlaceholderText } = render(<Note />)
    input = getByPlaceholderText('Your note goes here')
  })

  it('renders an input field with placeholder `Your note goes here`', () => {
    expect(input).toMatchSnapshot()
  })
})
