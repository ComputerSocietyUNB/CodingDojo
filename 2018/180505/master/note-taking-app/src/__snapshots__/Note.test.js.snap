// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Note renders an input field with placeholder \`Your note goes here\` 1`] = `
<input
  type="text"
  placeholder="Your note goes here"
  value=""
/>
`;
